client/packages/system/src/Encounter/DetailView/Toolbar.tsx

```diff
@@ -10,25 +10,25 @@ import {
   DateUtils,
   TimePickerInput,
   UserIcon,
+  useFormatDateTime,
 } from '@openmsupply-client/common';
-import { EncounterFragment, useEncounter } from '@openmsupply-client/programs';
+import { EncounterFragment } from '@openmsupply-client/programs';
 import { getClinicianName } from '../../Patient/Encounter';
+import { PatientFragment } from '../../Patient/api/operations.generated';
 
 const Row = ({ label, Input }: { label: string; Input: ReactNode }) => (
   <InputWithLabelRow labelWidth="90px" label={label} Input={Input} />
 );
 interface ToolbarProps {
   onChange: (patch: Partial<EncounterFragment>) => void;
+  patient: PatientFragment;
+  encounter: EncounterFragment;
 }
-export const Toolbar: FC<ToolbarProps> = ({ onChange }) => {
-  const id = useEncounter.utils.idFromUrl();
-  const { mutate: fetchEncounter, data: encounter } =
-    useEncounter.document.byIdPromise(id);
+export const Toolbar: FC<ToolbarProps> = ({ patient, encounter, onChange }) => {
```

Comment:
Rather than each sub-component doing its own queries, I thought it's better if the parent ("Details") component fetches the encounter and patient and passes them down to the Toolbar and Sidepanel (and Footer too if we need).

Having said that, if we include the Patient node along with the Enrolment node in https://github.com/openmsupply/open-msupply/issues/1155 then we can just fetch everything in the one query.

Reply:
I think react-query already takes care of the querying caching and I think we don't have to worry about calling a query twice from different component (as long as they use `useQuery`)

Reply:
Yeah, true. But one advantage of querying in the parent is that you can conditionally render the whole child component based on whether the query is resolved and that makes the logic _inside_ the child much tidier as you don't have to check for `encounter === undefined` all the time -- it's guaranteed to be there.

Reply:
> > I wasn't able to edit the note though...
> 
> Hmm, that's odd. What happens, exactly? (It's working fine for me -- did you reset the database?)

Just tried again its saved but if nothing else in the encounter has changed then the save button is disabled and I can't save the encounter... So if I only change the note I can't save it.